.github/scripts/determine-initial-guest-os-versions.py

```diff
@@ -0,0 +1,193 @@
+                    )
+                    continue
+                else:
+                    eprint_fmt(
+                        "Version %s targeting subnet %s in rollout %s is %s, taking into account",
+                        subnet["git_revision"],
+                        subnet["subnet_id"],
+                        rollout["name"],
+                        subnet["state"]
+                    )
+                t = batch.get("end_time") or batch.get("actual_start_time") or batch["planned_start_time"]
+                if subnet["subnet_id"] not in subnet_to_revision:
+                    subnet_to_revision[subnet["subnet_id"]] = []
+                subnet_to_revision[subnet["subnet_id"]].append((t, subnet["git_revision"]))
+
+    # Now we have a list of subnets associated with each
+    # Git revision coupled with the putative date or actual
+    # finish date for the revision.  Let's fish the latest
+    # revision for each subnet, and get that.
+    return list(set([
+        [revision for unused_date, revision in sorted(datestring_revision_tuple)][-1]
+        for datestring_revision_tuple in subnet_to_revision.values()
+    ]))
+
+def fetch_versions_from_public_dashboard(): # type: () -> list[str] | None
+    """
+    Fetch data from public dashboard
+
+    Panics if the parsed data is not in the expected format.
+    Returns an empty list if the action is retriable.
+    """
+    url = PUBLIC_DASHBOARD_ENDPOINT
+    try:
+        data = cast(PDSubnetsResponse, request_json(url))
+    except Exception as e:
+        eprint(f"Error fetching / decoding data from {url}: {e}.  Returning no versions.")
+        return []
+
+    subnets = data["subnets"]
+    versions = set()
+    for subnet in subnets:
+        try:
+            latest_replica_version = list(
+                sorted(
+                    [r for r in subnet["replica_versions"] if r.get("executed_timestamp_seconds")],
+                    key=lambda rr: rr.get("executed_timestamp_seconds") or 0 # the or 0 to satisfy py3.8 typechecking
+                )
+            )[-1]
+            versions.add(latest_replica_version["replica_version_id"])
+        except IndexError:
+            raise RuntimeWarning("Subnet %s does not have any executed version proposals" % subnet["subnet_id"])
+
+    return list(versions)
+
+def main():
+    unique_versions = fetch_versions_from_rollout_dashboard()
+    if not unique_versions:
+        eprint("No active rollouts found, will use versions from public dashboard")
+        unique_versions = fetch_versions_from_public_dashboard()
+
+    if not unique_versions:
+        # At this moment if we don't have any starting version we cannot proceed
+        raise RuntimeError(f"Didn't find any versions from:\n\t1. {ROLLOUT_DASHBOARD_ENDPOINT}\n\t2. {PUBLIC_DASHBOARD_ENDPOINT}")
+    eprint(f"Will qualify, starting from versions: {json.dumps(unique_versions)}")
+    matrix = {
+        "versions": unique_versions
```

Comment:
A typo here. It should be `version`, not `versions`

Reply:
Oh.